moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useCsvExport.js

```diff
@@ -87,49 +154,64 @@ export const useCsvExport = ({
   query,
   exportConfig,
   queryTableName,
-  fetchPolicy,
+  setDownloadingDialogOpen,
+  setDownloadOptionsDialogOpen,
+  columnDownloadOption,
+  setColumnDownloadOption,
+  visibleColumns,
 }) => {
+  const client = useApolloClient();
+
   /**
-   * When True, the download csv dialog is open.
-   * @type {boolean} dialogOpen
-   * @function setDialogOpen - Sets the state of dialogOpen
-   * @default false
+   * Open the column download options dialog when export button is clicked
    */
-  const [dialogOpen, setDialogOpen] = useState(false);
+  const handleExportButtonClick = () => {
+    setDownloadOptionsDialogOpen(true);
+  };
 
   /**
-   * Instantiates getExport and data variables
-   * @function getExport - It is called to load the data
-   * @property {object} data - The data as retrieved from query (if available)
+   * Closes the download options dialog when user clicks away or on cancel button
    */
-  const [getExport] = useLazyQuery(query, {
-    fetchPolicy: fetchPolicy,
-  });
+  const handleOptionsDialogClose = () => {
+    setDownloadOptionsDialogOpen(false);
+  };
 
   /**
-   * Handles export button (to open the csv download dialog)
+   * Updates the column download option based on user radio button selection
    */
-  const handleExportButtonClick = () => {
-    setDialogOpen(true);
-
-    // Fetch data and format, parse, and download CSV when returned
-    getExport().then(({ data }) => {
-      const formattedData = formatExportData(
-        data[queryTableName],
-        exportConfig
-      );
-      const csvString = Papa.unparse(formattedData, { escapeFormulae: true });
-      downloadFile(csvString);
-      handleDialogClose();
-    });
+  const handleRadioSelect = (e) => {
+    setColumnDownloadOption(e.target.value);
   };
 
   /**
-   * Handles the closing of the dialog
+   * Downloads the csv and opens the downloading dialog when continue button is clicked
    */
-  const handleDialogClose = () => {
-    setDialogOpen(false);
+  const handleContinueButtonClick = () => {
+    setDownloadOptionsDialogOpen(false);
+    setDownloadingDialogOpen(true);
+    // Fetch data and format, parse, and download CSV when returned
+    client
+      .query({
+        query: query,
+        fetchPolicy: "network-first",
+      })
```

Comment:
This works so well! I see the query only happen once (and only once!) click and not when toggling the radios. 🥳